beacon-chain/rpc/eth/beacon/blocks.go

```diff
@@ -67,6 +67,10 @@ func (bs *Server) GetBlockHeader(ctx context.Context, req *ethpbv1.BlockRequest)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not determine if block root is canonical: %v", err)
 	}
+	isOptimistic, err := bs.HeadFetcher.IsOptimistic(ctx)
```

Comment:
This is wrong. 

The spec says: 
```
- description: "True if the response references an unverified execution payload. Optimistic information may be invalidated at a later time. If the field is not present, assume the False value."
```

The `isOptimistic` should reference to the responded block header not to the current head. Same for other places

Reply:
You're right. I changed the code to use `IsOptimisticForRoot` everywhere. The only two places where we look at the head are `head` and `chain_reorg` events.